torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -516,6 +516,66 @@ class IrParser {
           });
     }
 
+    {
+      auto ptr_op = getOperatorForLiteral(
+          "aten::native_dropout(Tensor input, float p, float scale, bool train) -> (Tensor, Tensor)");
+      registerParseRule(
+          ptr_op,
+          [](const Node* node,
+             std::unordered_map<size_t, CgValue>& value_map) -> void {
+            auto input = value_map[node->input(0)->unique()];
+            auto prob = value_map[node->input(1)->unique()];
+            auto scale = value_map[node->input(2)->unique()];
+            auto train = constant_as<bool>(node->input(3));
+
+            TORCH_INTERNAL_ASSERT(train.has_value() and train.value(), 
```

Comment:
Wasn't dropout train==False just a tensor forwarding?
It does introduce alias (which breaks the schema), but we can just forward the tensor as-is.

Checkout `_grad_sum_to_size` where `!size_to->empty()`

Reply:
In this situation, we shouldn't see `train==False`.  The `aten::dropout` signature is where we get `train==False` and that is where tensor forwarding would apply.

Reply:
Oops, I wasn't careful enough ealier.

It is dangerous for us to assume that `dropout` is used for inference and `native_dropout` is used for training, even though this is totally the case if we use `torch.nn.Dropout`.

`torch.nn.functional.dropout` is exposed at user level. So it is totally possible for them to give arbitrary argument for `training` with combination to `requires_grad_` on input.